### PR TITLE
Issue #2170: New check for getContractAt()

### PR DESCRIPTION
**Pull request by SfyMantissa**
- [x] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.

---

New feature. Was previously discussed in [Issue #2170](https://github.com/NomicFoundation/hardhat/issues/2170).
Adds a check for input address to `getContractAt()` — now it MUST have code (be a contract), otherwise function throws.

What was done:

- [x] Changes to the module itself
- [x] Test update
- [x] README update 

All tests run fine:

image

**Let me know if anything else is needed for submission :)**

### CODE REVIEW

File: packages/hardhat-ethers/README.md [hunk 1]
```diff
@@ -79,6 +79,8 @@ The [`Contract`s](https://docs.ethers.io/v5/single-page/#/v5/api/contract/contra
 
 If there is no signer available, `getContractAt` returns [read-only](https://docs.ethers.io/v5/single-page/#/v5/api/contract/contract/-%23-Contract--readonly) contracts.
 
+If a non-contract address is provided to `getContractAt`, an error will be thrown.
```
Comment (feuGeneA) [inline packages/hardhat-ethers/README.md:82]:
```suggestion
If the address provided to `getContractAt` is not the address of a contract account, an error will be thrown.
```

File: packages/hardhat-ethers/src/internal/helpers.ts [hunk 2]
```diff
@@ -303,6 +303,13 @@ export async function getContractAt(
   address: string,
   signer?: ethers.Signer
 ) {
+  if ((await hre.ethers.provider.getCode(address)) === "0x") {
+    throw new NomicLabsHardhatPluginError(
+      pluginName,
+      `${address} is a non-contract address.`
```
Comment (feuGeneA) [inline packages/hardhat-ethers/src/internal/helpers.ts:309]:
```suggestion
      `${address} is not a contract account.`
```

File: packages/hardhat-ethers/test/index.ts [hunk 3]
```diff
@@ -710,6 +710,14 @@ describe("Ethers plugin", function () {
         });
 
         describe("by name and address", function () {
+          it("Should throw if address does not belong to a contract", async function () {
+            const address = await signers[0].getAddress();
+            return assert.isRejected(
+              this.env.ethers.getContractAt("Greeter", address),
+              `${address} is a non-contract address.`
```
Comment (feuGeneA) [inline packages/hardhat-ethers/test/index.ts:717]:
```suggestion
              `${address} is not a contract account.`
```

**Review by feuGeneA — CHANGES_REQUESTED** [review]
Thank you for this!

**Comment by SfyMantissa** [comment]
Hi!
I accepted the proposed changes to the description strings.
Is there anything else needed to accept this PR?

**Comment by SfyMantissa** [comment]
Hi!
Sorry for bothering, is there anything else that needs to be polished/added?

**Review by feuGeneA — APPROVED** [review]

**Pull request merged**